frontend/src/lib/components/EventPropertyTabs/EventPropertyTabs.tsx

```diff
@@ -0,0 +1,207 @@
+import { eventPropertyFilteringLogic } from 'lib/components/EventPropertyTabs/eventPropertyFilteringLogic'
+import { INTERNAL_EXCEPTION_PROPERTY_KEYS } from 'products/error_tracking/frontend/utils'
+import { dayjs } from 'lib/dayjs'
+
+export interface TabContentComponentFnProps {
+    event: EventType | RecordingEventType
+    properties: Record<string, any>
+    promotedKeys?: string[]
+    tabKey: EventPropertyTabKey
+}
+
+type EventPropertyTabKey =
+    | 'properties'
+    | 'flags'
+    | 'image'
+    | 'elements'
+    | '$set_properties'
+    | '$set_once_properties'
+    | 'raw'
+    | 'conversation'
+    | 'exception_properties'
+    | 'error_display'
+    | 'debug_properties'
+    | 'metadata'
+
+export const EventPropertyTabs = ({
+    event,
+    tabContentComponentFn,
+    ...lemonTabsProps
+}: {
+    event: EventType | RecordingEventType
+    tabContentComponentFn: (props: TabContentComponentFnProps) => JSX.Element
+    dataAttr?: LemonTabsProps<EventPropertyTabKey>['data-attr']
+    size?: LemonTabsProps<EventPropertyTabKey>['size']
+    barClassName?: LemonTabsProps<EventPropertyTabKey>['barClassName']
+}): JSX.Element => {
+    const isAIGenerationEvent = event.event === '$ai_generation'
+    const isAIEvent = isAIGenerationEvent || event.event === '$ai_span' || event.event === '$ai_trace'
+
+    const isErrorEvent = event.event === '$exception'
+
+    const { filterProperties } = useValues(eventPropertyFilteringLogic)
+
+    const [activeTab, setActiveTab] = useState<EventPropertyTabKey>(
+        isAIEvent ? 'conversation' : isErrorEvent ? 'error_display' : 'properties'
+    )
+
+    const promotedKeys = POSTHOG_EVENT_PROMOTED_PROPERTIES[event.event]
+
+    let properties = {}
+    const featureFlagProperties = {}
+    const errorProperties = {}
+    const debugProperties = {}
+    let setProperties = {}
+    let setOnceProperties = {}
+
+    for (const key of Object.keys(event.properties)) {
+        if (!CORE_FILTER_DEFINITIONS_BY_GROUP.events[key] || !CORE_FILTER_DEFINITIONS_BY_GROUP.events[key].system) {
+            if (CORE_FILTER_DEFINITIONS_BY_GROUP.event_properties[key]?.used_for_debug) {
+                debugProperties[key] = event.properties[key]
+            } else if (key.startsWith('$feature') || key === '$active_feature_flags') {
+                featureFlagProperties[key] = event.properties[key]
+            } else if (key === '$set') {
+                setProperties = event.properties[key] ?? {}
+            } else if (key === '$set_once') {
+                setOnceProperties = event.properties[key] ?? {}
+            } else if (INTERNAL_EXCEPTION_PROPERTY_KEYS.includes(key)) {
+                errorProperties[key] = event.properties[key]
+            } else {
+                properties[key] = event.properties[key]
+            }
+        }
+    }
+    properties = {
+        ...('timestamp' in event ? { $timestamp: dayjs(event.timestamp).toISOString() } : {}),
+        ...filterProperties(properties),
+    }
```

Comment:
I had duplicated changes between `ItemEvent` and `EventsDisplay` about 4 times now

This was the straw that broke the camel's back

There's a little yuckiness in that they want to use slightly different content in the tabs themselves

But I think much nicer like this with the differences so clearly called out in the consuming components